includes/admin/models/class-rop-queue-model.php

```diff
@@ -300,6 +300,60 @@ function ( $a, $b ) {
 		return $ordered;
 	}
 
+	/**
+	 * Method to build the queue for posts to be published on update/create.
+	 *
+	 * @access  public
+	 * @return array
+	 */
+	public function build_queue_publish_now() {
+		$settings_model     = new Rop_Settings_Model();
+		$post_types         = wp_list_pluck( $settings_model->get_selected_post_types(), 'value' );
+
+		// fetch all post_types that were modified in the last 30 seconds and need to be published now.
+		$query              = new WP_Query(
```

Comment:
let's move the posts selection query into the post selector model, this class is used only for queue related actions, not posts filtering/querying. 

I see some problems here on how you fetch the posts: 
1) You don't delete the post meta `rop_publish_now` after the post has been shared, keeping it might cause issues and I don't see a reason why we should have it after the publish occurred. 
2) I don't really understand why you don't pass when creating the publish now cron, the post id which was modified and do this query? Is there any reason that made you use this approach?

Reply:
`rop_publish_now` indicates that the publish now has been selected. This will be used even when the post is updated from the backend without actually saving the post from the UI. i.e. if someone fires a `wp_update_post`. I don't think we need to clear this.

I don't understand your 2nd point.

Reply:
@contactashish13 for the first point what will happen in this case:

1) The user wants to share on social networks a post on publishing. User comes back 1 week later, edit the same post but don't want to share on social network. What will happen? 
2) If i edit 2 posts within 30 seconds and i chose to share via rop just one, what will happen ?

As my 2nd point in my initial request, my idea is: 
1) Here -> https://github.com/Codeinwp/tweet-old-post-pro/pull/108/files#diff-f16969e9a5640144a76a015708b35068R242 why don't you pass the id of the post and use it in the sharing action,  instead of querying anymore the posts updated in the last 30s and avoid building a special queue for this action ?

Reply:
Please check the logic for determining the posts that need to be shared. I'm checking for posts modified in the last 30 seconds that contain the `rop_publish_now` flag. For posts that should not be share, why would a user check that checkbox? 

As for
> why don't you pass the id of the post and use it in the sharing action

Please point me to the piece of code that you think can do this as I don't see any portion of code that supports this. The existing code does not have the ability to accept specific post ids.

Reply:
@contactashish13 let's asume you send to the publish now cron the post id and account keys. 
```
$account_data = $services_model->find_account( $account );
						try {
							$service = $service_factory->build( $account_data['service'] );
							$service->set_credentials( $account_data['credentials'] );
							foreach ( $posts as $post ) {
								$post_data = $queue->prepare_post_object( $post, $account );
								$logger->info( 'Posting', array( 'extra' => $post_data ) );
								$service->share( $post_data, $account_data );
							}
						} catch ( Exception $exception ) {
							$error_message = sprintf( Rop_I18n::get_labels( 'accounts.service_error' ), $account_data['service'] );
							$logger->alert_error( $error_message . ' Error: ' . $exception->getTrace() );
						}
```

won't this work to share?

Reply:
@selul this approach would create multiple events, one for each post that needs to be shared and I did not prefer that as it would clutter the cron.

Reply:
we can send all the accounts key along with the post id when creating the cron. 
In this way, we follow the same pattern for the scheduled tweets.
Don't you think will be better than querying the last 30s posts?

Reply:
@selul I thought my approach was more inline with scheduled tweets considering the below points :)

1. There is only one event (with your suggested approach there will be multiple events)
2. The cron builds the queue to work on (in your suggested approach it will be be given the queue)

But it can be changed if you think that is a better implementation.

Reply:
Thinking more about this, I think you are right, we can keep it like this and see if there any issues in the future.